Xamarin.Essentials/SecureStorage/SecureStorage.android.cs

```diff
@@ -202,7 +202,12 @@ public byte[] Encrypt(string data)
             sr.NextBytes(iv);
 
             var cipher = Cipher.GetInstance(cipherTransformationSymmetric);
-            cipher.Init(CipherMode.EncryptMode, key, new GCMParameterSpec(128, iv));
+
+            // Use GCMParameterSpec on newer API levels and IvParameterSpec on older
+            if (Platform.HasApiLevel(BuildVersionCodes.M) && !alwaysUseAsymmetricKey)
```

Comment:
Is this even possible to reach here? On M, the library will use the symmetric key. If this is an issue on 4.4/KitKat, do you mean android L?

Reply:
This code is the bit that encrypts the actual data using the AES key (which comes either from Keystore directly on >= M, or from the preferences (which wrap it in an RCA asymmetric key).  So this code path should run on every api level.